src/ir/proto/sql/decode_test.cc

```diff
@@ -113,4 +114,96 @@ absl::string_view kLiteralStrs[] = {
 INSTANTIATE_TEST_SUITE_P(
     DecodeLiteralTest, DecodeLiteralTest, ValuesIn(kLiteralStrs));
 
+TEST(DecodeSourceTableColumnExprTest, DecodeSourceTableColumnExprTest) {
+  const std::string kTextproto =
+      R"(id: 1 name: "alias" source_table_column: { column_path: "table1.col" })";
+  IRContext ir_context;
+  DecoderContext decoder_context(ir_context);
+  Expression expr;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(kTextproto, &expr))
+    << "Could not decode expr";
+  const ir::Value &result = DecodeExpression(expr, decoder_context);
+  const value::StoredValue *stored_value = result.If<value::StoredValue>();
+  EXPECT_THAT(stored_value, testing::NotNull());
+  const Storage &storage = stored_value->storage();
+  EXPECT_EQ(storage.name(), "table1.col");
+  EXPECT_EQ(storage.type().type_base().kind(),
+            types::TypeBase::Kind::kPrimitive);
```

Comment:
seems like a repeat of the `DecodeSourceColumnTableTest`s. Is there anyway to reuse them via typed parameterized or value parameterized tests?

Reply:
I'll see what I can do here.

Reply:
I think the source of this duplication is that I have been testing the variant decoding methods as if they were public-facing methods. But really, they shouldn't be: the only public-facing method will be the method that decodes an `Expression`, and the others should be `static` helper functions for decoding an `Expression`. I'll make those methods not part of the public API, which will remove the need to test them individually, and will test them only through `DecodeExpression`.

Reply:
SG. Please do that.

Reply:
Done.